src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDsa.cs

```diff
@@ -107,6 +186,28 @@ public virtual bool TrySignData(ReadOnlySpan<byte> data, Span<byte> destination,
             return false;
         }
 
+        public bool TrySignData(ReadOnlySpan<byte> data, Span<byte> destination, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat, out int bytesWritten)
+        {
+            return TrySignDataCore(data, destination, hashAlgorithm, signatureFormat, out bytesWritten);
+        }
+
+        protected virtual bool TrySignDataCore(ReadOnlySpan<byte> data, Span<byte> destination, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat, out int bytesWritten)
+        {
+            if (string.IsNullOrEmpty(hashAlgorithm.Name))
+            {
+                throw new ArgumentException(SR.Cryptography_HashAlgorithmNameNullOrEmpty, nameof(hashAlgorithm));
+            }
+
+            if (TryHashData(data, destination, hashAlgorithm, out int hashLength) &&
+                TrySignHashCore(destination.Slice(0, hashLength), destination, signatureFormat, out bytesWritten))
```

Comment:
I don't know that we want to depend on the implementation of TrySignHashCore to support signing in-place (that it can, without fail, write a signature over top of the hash value memory).

Please hash into a temporary location (stackalloc/rent).  This also applies to DSA if I missed it there.

Reply:
I was originally planning to do that but I noticed we already did that before and followed the pattern but will fix